### PR TITLE
feat: Contract deployment module should use the warp gw address passe…

### PR DESCRIPTION
**Pull request by ppedziwiatr**
…d in the configuration #220

### CODE REVIEW

File: src/utils/utils.ts [hunk 1]
```diff
@@ -86,3 +86,20 @@ export function bufToBn(buf: Buffer) {
...
+export async function getJsonResponse<T>(response: Promise<Response>): Promise<T> {
+  let r;
+  try {
+    r = await response;
+  } catch (e) {
+    throw new Error(`Error while communicating with gateway: ${JSON.stringify(e)}`);
+  }
+
+  if (!r?.ok) {
+    const text = await r.text();
```
Comment (asiaziola) [inline src/utils/utils.ts:99]: works quite ok for errors thrown through the koa context like this one:
`ctx.throw(400, 'Error message');`

the error thrown by sdk:
`Error: 400: {"message":{"message":"Contract data item binary is not valid."}}`

for the errors thrown through regular Error object:
`throw new Error('Error text.')`

the error thrown by sdk:
`Error: 500: {"message":{}}`

I'm not sure if we could always pass error message in the ctx.body when error is thrown? like this:
`ctx.body = e.message`

File: src/core/modules/impl/WarpGatewayContractDefinitionLoader.ts [hunk 2]
```diff
@@ -67,20 +65,11 @@ export class WarpGatewayContractDefinitionLoader implements DefinitionLoader {
...
-        .catch((error) => {
-          if (error.body?.message) {
-            this.rLogger.error(error.body.message);
-          }
-          throw new Error(
-            `Unable to retrieve contract data. Warp gateway responded with status ${error.status}:${error.body?.message}`
-          );
-        });
+      const baseUrl = stripTrailingSlash(this._warp.gwUrl());
+      const result: ContractDefinition<State> = await getJsonResponse(
```
Comment (asiaziola) [inline src/core/modules/impl/WarpGatewayContractDefinitionLoader.ts:69]: await getJsonResponse<ContractDefinition<State>>

File: src/utils/utils.ts (hunk 1 above)
Comment (ppedziwiatr) [reply src/utils/utils.ts:99]: the last case is be fixed by the changes in the GW - https://github.com/warp-contracts/gateway/pull/161

File: src/core/modules/impl/WarpGatewayContractDefinitionLoader.ts (hunk 2 above)
Comment (ppedziwiatr) [reply src/core/modules/impl/WarpGatewayContractDefinitionLoader.ts:69]: nope.

**Pull request merged**